### ISSUE TITLE
Kotlin back tick function names don't highlight as standard function

### ISSUE THREAD

**Issue by c1phr** [jneen/rouge#906]
In Kotlin, functions (particularly test methods) can be named using "natural language" by wrapping the name in backticks, see [convention docs](https://kotlinlang.org/docs/reference/coding-conventions.html#naming-rules). When highlighting Kotlin code with rogue, backticks get highlighted and the string inside isn't colored as a normal method name as it probably should be.

Expected result: Method names with backticks and spaces should be highlighted similar to "normal" methods.

Snippet to recreate:

Standard Kotlin method (working):

`fun handlerShouldSaveRecordToDbOnUpdatedEventProcessed() {}`

Backtick method naming (not working):

``fun `handler should save record to db on updated event processed`() {}``

**Comment by mkobit** [comment]
haven't tried it, but I bet the same is true for other identifiers

```
class `Why would you name your class this?`
```

```
val `this is a bad idea` = "hello"
```

**Comment by rongi** [comment]
Same issue as this one https://github.com/jneen/rouge/issues/791

**Comment by lordcodes** [comment]
This issue is fixed by #996

**Comment by lordcodes** [comment]
This is no longer working, so either there has been a regression with a later change or it wasn't successfully fixed at the time. I submitted a few PRs around the same time, so it could be that they conflicted with each other somehow.

Will need to investigate more why it isn't working.
